builtin/providers/terraform/resource_state.go

```diff
@@ -62,7 +62,11 @@ func resourceRemoteStateRead(d *schema.ResourceData, meta interface{}) error {
 
 	var outputs map[string]string
 	if !state.State().Empty() {
-		outputs = state.State().RootModule().Outputs
+		outputValueMap := make(map[string]string)
+		for key, output := range state.State().RootModule().Outputs {
+			//This is ok for 0.6.17 as outputs will have been strings
+			outputValueMap[key] = output.Value.(string)
+		}
```

Comment:
outputValueMap looks like it's thrown away here as a local variable?

I think this should just be populating `outputs`, the var from the higher level scope.